stages/org.osbuild.users

```diff
@@ -70,6 +70,10 @@ SCHEMA = """
             "items": {
               "type": "string"
             }
+          },
+          "expiredate": {
```

Comment:
I did a quick grep around and couldn't really find any other stages where we take dates. I'd like to do two things here:

1. A standardized format for the input, let's pick ISO 8601: `YYYY-MM-DDT[HH[:MM[:SS[.mmm[uuu]]]]][+HH:MM]` as it is widely supported (and easy to do in Python).
2. A regular expression that validates the format in the schema.

We can then reformat the expire date in the required `YYYY-MM-DD` format down below.

Reply:
This is fine with me. When we convert the ISO date to `YYYY-MM-DD` do we want to truncate or round to the nearest day?

Reply:
Let's truncate and document that in the description. The behavior is then that the entire day the expire date falls on is the day the account expires.

Reply:
Also, my understanding is that the information stored in `/etc/shadow` is a number of days since Jan 1 1970, so I'm assuming this is basically a unix timestamp (but in days rather than seconds). However, it is not clear to me how `useradd` converts YYYY-MM-DD to a unix timestamp. It must assume a timezone since no timezone is provided, but which one does it assume? It could be UTC since the unix timestamp is utc, or it could be the timezone of the build machine. Could it even be the timezone of the final image? It's not clear to me which one of these options is correct. When I took a peek at the the [implementation of `useradd`](https://github.com/shadow-maint/shadow/blob/master/src/useradd.c#L1291) to try to figure this out, I followed it to [this comment](https://github.com/shadow-maint/shadow/blob/master/lib/strtoday.c#L48-L49) that explains that pure numerical values are interpreted as days since the epoch. This means if I pass `--expiredate 12345` to `useradd`, that is the exact number that appears in `/etc/shadow`. This let's me bypass the whole timezone question, BUT this does not appear to be documented behavior for `useradd`/`usermod`. Did you have any thoughts on how this should be handled?

Reply:
I agree.  It keeps the stage simple and easier to understand and higher level tools can do the necessary computation if needed.  Also we can always change the field to a union type if we need to (though that's always a tiny bit annoying).

Reply:
Oops, sorry I missed that bit of documentation! I'll make it an integer :+1:

Reply:
This parameter is now an `integer`.